src/database-gdbm.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * GDBM database interface for Minetest
+ *
+Copyright (C) 2017 Maciej 'agaran' Pijanka <agaran@pld-linux.org>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "config.h"
+
+#if USE_GDBM
+#include "database-gdbm.h"
+
+#include "log.h"
+#include "filesys.h"
+#include "exceptions.h"
+#include "settings.h"
+#include "porting.h"
+#include "util/string.h"
+
+#include <netinet/in.h>
+
+#include <cassert>
+
+Database_Gdbm::Database_Gdbm(const std::string &savedir) :
+	m_savedir(savedir),
+	m_database(NULL)
+{
+	std::string dbfile = m_savedir + DIR_DELIM + "map.gdbm";
+
+	if (!fs::CreateAllDirs(m_savedir)) {
+		infostream << "Database_GDBM: Failed to create directory \""
+			<< m_savedir << "\"" << std::endl;
+		throw FileNotGoodException("Failed to create database "
+				"save directory");
+	}
+
+	char *name = (char *)malloc(dbfile.length()+1);
+	memcpy(name, dbfile.c_str(), dbfile.length());
+	name[dbfile.length()] = '\0';
```

Comment:
wtf just read the docs
`GDBM_FILE gdbm_open (const char *name, int block_size,   int flags, int mode, void (*fatal_func)(const char *))`

you can reduce it to `dbfile.c_str()`...